lib/localparticipant.js

```diff
@@ -316,6 +317,14 @@ class LocalParticipant extends Participant {
           self._addTrackPublication(localTrackPublication);
         }
 
+        localTrackPublication.on('warning', twilioWarningName => {
+          self.emit('trackWarning', twilioWarningName, localTrackPublication);
+        });
+
+        localTrackPublication.on('warningsCleared', () => {
+          self.emit('trackWarningsCleared', localTrackPublication);
+        });
```

Comment:
```suggestion
         const events =  [
           'trackWarning',
           'trackWarningsCleared'
         ];
         const reemitEvent = (...args) =>
           self.emit(event, ...[...args, participant]);
         events.forEach(event =>
           localTrackPublication.on(event, reemitEvent));
         /* Remove reemitEvent listener when localTrackPublication.unpublish() is called */
```

Reply:
@manjeshbhargav the suggestion won't work because the events that we're reemitting are different. I think we should keep it as is for this case to make it more readable.